userspace/libscap/scap.h

```diff
@@ -1154,6 +1154,109 @@ uint64_t scap_get_driver_api_version(scap_t* handle);
  */
 uint64_t scap_get_driver_schema_version(scap_t* handle);
 
+#ifndef SCAP_HANDLE_T
+#define SCAP_HANDLE_T void
+#endif
+
+#ifndef SCAP_OPEN_PARAM_T
+#define SCAP_OPEN_PARAM_T void
+#endif
+
+/*
+ * magic constants, matching the kmod-only ioctl numbers defined as:
+#define PPM_IOCTL_MAGIC	's'
+#define PPM_IOCTL_MASK_ZERO_EVENTS _IO(PPM_IOCTL_MAGIC, 5)
+#define PPM_IOCTL_MASK_SET_EVENT   _IO(PPM_IOCTL_MAGIC, 6)
+#define PPM_IOCTL_MASK_UNSET_EVENT _IO(PPM_IOCTL_MAGIC, 7)
+ */
+enum scap_eventmask_op {
+	SCAP_EVENTMASK_ZERO = 0x7305,
+	SCAP_EVENTMASK_SET = 0x7306,
+	SCAP_EVENTMASK_UNSET = 0x7307,
+};
+
+enum scap_setting {
+	SCAP_SAMPLING_RATIO,
+	SCAP_TRACERS_CAPTURE,
+	SCAP_PAGE_FAULTS,
+	SCAP_SNAPLEN,
+	SCAP_EVENTMASK,
+	SCAP_DYNAMIC_SNAPLEN,
+	SCAP_SIMPLEDRIVER_MODE,
+	SCAP_FULLCAPTURE_PORT_RANGE,
+	SCAP_STATSD_PORT,
+};
+
+// this is passed by value everywhere so keep it small
+// *temporarily*, it contains a pointer to the full scap handle
+// eventually, it will contain a pointer to a struct containing
+// only the engine-specific bits
+struct scap_engine_handle {
+	struct scap* m_handle;
+};
```

Comment:
From the comment, it seems like this struct, that now contains a pointer to the global scap handler, will contain a single pointer to the engine-specific context. My question is, what's the advantage of having a structure with a single member instead of a single opaque pointer?

Reply:
IIRC it made some type safety easier (fewer casts between `void*` and a concrete `T*`). But I agree it's not too pretty so I'll try and see if just using a pointer works.

(note: in the public header, we don't really know what the type of the opaque handle is--that's the point of being an opaque handle :D)